wpilibj/src/athena/java/edu/wpi/first/wpilibj/DriverStation.java

```diff
@@ -230,27 +287,39 @@ private void reportJoystickUnpluggedWarning(String message) {
    * @param axis  The analog axis value to read from the joystick.
    * @return The value of the axis on the joystick.
    */
-  public synchronized double getStickAxis(int stick, int axis) {
-    if (stick < 0 || stick >= kJoystickPorts) {
-      throw new RuntimeException("Joystick index is out of range, should be 0-5");
-    }
+  public double getStickAxis(int stick, int axis) {
+    boolean lockEntered = false;
+    try {
+      m_joystickMutex.lock();
+      lockEntered = true;
+      if (stick < 0 || stick >= kJoystickPorts) {
+        throw new RuntimeException("Joystick index is out of range, should be 0-5");
```

Comment:
Yeah. The try finally is actually what the compiler expands synchronized(m_joystickMutex) to. It doesn't do the if (lockEntered) thing, but thats the only way to leave a lock early.

Also fixed the rest of the line comments above.